snapcraft/parts/plugins/matter_sdk_plugin.py

```diff
@@ -0,0 +1,180 @@
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The matter SDK plugin."""
+import os
+from typing import Any, Dict, List, Set, cast
+
+from craft_parts import infos, plugins
+from overrides import overrides
+
+# The repository where the matter SDK resides.
+MATTER_SDK_REPO = "https://github.com/project-chip/connectedhomeip"
+
+
+class MatterSdkPluginProperties(plugins.PluginProperties, plugins.PluginModel):
+    """The part properties used by the matter SDK plugin."""
+
+    matter_sdk_version: str
+
+    @classmethod
+    @overrides
+    def unmarshal(cls, data: Dict[str, Any]) -> "MatterSdkPluginProperties":
+        """Populate class attributes from the part specification.
+
+        :param data: A dictionary containing part properties.
+
+        :return: The populated plugin properties data object.
+
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = plugins.extract_plugin_properties(
+            data,
+            plugin_name="matter-sdk",
+            required=["matter_sdk_version"],
+        )
+        return cls(**plugin_data)
+
+
+class MatterSdkPlugin(plugins.Plugin):
+    """A plugin for matter SDK project.
+
+    This plugin uses the common plugin keywords.
+    For more information check the 'plugins' topic.
+
+    Additionally, this plugin uses the following plugin-specific keywords:
+        - matter-sdk-version
+          (str, no default)
+          The matter SDK version to use for the build.
+    """
+
+    properties_class = MatterSdkPluginProperties
+
+    def __init__(
+        self,
+        *,
+        properties: plugins.PluginProperties,
+        part_info: infos.PartInfo,
+    ) -> None:
+        super().__init__(properties=properties, part_info=part_info)
+
+        self.matter_sdk_dir = part_info.part_build_dir
+        self.snap_arch = os.getenv("SNAP_ARCH")
+
+    @overrides
+    def get_pull_commands(self) -> List[str]:
+        options = cast(MatterSdkPluginProperties, self._options)
+        commands = []
+
+        # Clone Matter SDK repository
+        commands.extend(
+            [
+                "if [ ! -d matter ]; then",
+                "    git init",
+                f"   git remote add origin {MATTER_SDK_REPO}",
+                f"   git fetch --depth 1 origin {options.matter_sdk_version}",
+                "    git checkout FETCH_HEAD",
+                "fi",
+            ]
+        )
+
+        # Checkout submodules for Linux platform
+        commands.extend(["scripts/checkout_submodules.py --shallow --platform linux"])
```

Comment:
You may want to do this in a subshell if it is under `matter` that was changed above
```suggestion
        commands.extend(["(cd matter && ./scripts/checkout_submodules.py --shallow --platform linux)"])
```

Reply:
Explained as above